src/App.tsx

```diff
@@ -15,48 +16,89 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  length = 'length',
+  alphabet = 'alphabet',
+  noSort = '',
+}
+
+function sortGoods(sortField: SortType, isReversed: boolean) {
+  const goods = [...goodsFromServer];
+
+  if (sortField) {
+    goods.sort((a, b) => {
+      switch (sortField) {
+        case SortType.length:
+          return a.length - b.length;
+        case SortType.alphabet:
+          return a.localeCompare(b);
+        default:
+          return 0;
+      }
+    });
+  }
+
+  return isReversed ? goods.reverse() : goods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState(SortType.noSort);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const goods = sortGoods(sortField, isReversed);
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={classNames('button is-info',
+            { 'is-light': sortField !== SortType.alphabet })}
+          onClick={() => setSortField(SortType.alphabet)}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={classNames('button is-success',
+            { 'is-light': sortField !== SortType.length })}
+          onClick={() => setSortField(SortType.length)}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={classNames('button is-warning',
+            { 'is-light': !isReversed })}
+          onClick={() => setIsReversed(!isReversed)}
         >
           Reverse
         </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {(sortField !== '' || isReversed) && (
```

Comment:
An empty string is a `falsy` value - and you use it directly in your sort method (ln 28)
Please be consistent with this